src/renderer/components/execute/conditions/UnitTestTab.tsx

```diff
@@ -0,0 +1,54 @@
+import { UnitTestType } from 'db/models/UnitTestModels';
+import { Button, Container, Row } from 'react-bootstrap';
+import useModal from 'renderer/hooks/useModal';
+import AddUnitTestModal from './AddUnitTestModal';
+import UnitTestList from './UnitTestList';
+
+interface IUnitTestsProps {
+  conditionList: Partial<UnitTestType>[];
+  setConditionList: (c: Partial<UnitTestType>[]) => void;
+}
+
+const UnitTestTab = ({ conditionList, setConditionList }: IUnitTestsProps) => {
+  const { isOpen, toggle } = useModal();
+
+  const deleteCondition = (condition: Partial<UnitTestType>) => {
+    const newList = conditionList.filter(
+      (c: Partial<UnitTestType>) => c !== condition
+    );
+    setConditionList(newList);
+  };
+
+  const addCondition = (condition: Partial<UnitTestType>) => {
```

Comment:
If you're adding directly to the conditionList, do you need to call setConditionList()

Reply:
Yes, setConditionList is a hook which when called React checks if the prop values changed and if so re-renders the component